metrics/prometheus.go

```diff
@@ -0,0 +1,261 @@
+				"connections",
+				"Count of total number of connectionsbackend get requests to Prebid Server labeled by status.",
+			),
+			ConnectionsErrors: newCounterVecWithLabels(cfg, registry,
+				"connection_error",
+				"Count the number of connection accept errors or connection close errors",
+				[]string{"connection_error"},
+			), // "connection_error" = {"accept", "close"}
+		},
+		ExtraTTL: &PrometheusExtraTTLMetrics{
+			ExtraTTLSeconds: newHistogram(cfg, registry,
+				"extra_ttl_seconds",
+				"Extra time to live in seconds specified",
+				cacheWriteTimeBuckts,
+			),
+		},
+	}
+	promMetrics.ExtraTTL.ExtraTTLSeconds.Observe(5000.00)
+
+	return promMetrics
+}
+
+/**************************************************
+ *	Helper Init functions
+ **************************************************/
+func newCounterVecWithLabels(cfg config.PrometheusMetrics, registry *prometheus.Registry, name string, help string, labels []string) *prometheus.CounterVec {
+	opts := prometheus.CounterOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+	}
+	counterVec := prometheus.NewCounterVec(opts, labels)
+	registry.MustRegister(counterVec)
+	return counterVec
+}
+
+func newSingleCounter(cfg config.PrometheusMetrics, registry *prometheus.Registry, name string, help string) prometheus.Counter {
+	opts := prometheus.CounterOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+	}
+	counter := prometheus.NewCounter(opts)
+	registry.MustRegister(counter)
+	return counter
+}
+
+func newHistogram(cfg config.PrometheusMetrics, registry *prometheus.Registry, name, help string, buckets []float64) prometheus.Histogram {
+	opts := prometheus.HistogramOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+		Buckets:   buckets,
+	}
+	histogram := prometheus.NewHistogram(opts)
+	registry.MustRegister(histogram)
+	return histogram
+}
+
+func newGaugeMetric(cfg config.PrometheusMetrics, registry *prometheus.Registry, name string, help string) prometheus.Gauge {
+	opts := prometheus.GaugeOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+	}
+	gauge := prometheus.NewGauge(opts)
+	registry.MustRegister(gauge)
+	return gauge
+}
+
+func newHistogramVector(cfg config.PrometheusMetrics, registry *prometheus.Registry, name, help string, labels []string, buckets []float64) *prometheus.HistogramVec {
+	opts := prometheus.HistogramOpts{
+		Namespace: cfg.Namespace,
+		Subsystem: cfg.Subsystem,
+		Name:      name,
+		Help:      help,
+		Buckets:   buckets,
+	}
+	histogram := prometheus.NewHistogramVec(opts, labels)
+	registry.MustRegister(histogram)
+	return histogram
+}
+
+/**************************************************
+ *	DEPECRATED Functions to record metrics
+ **************************************************/
+func (m PrometheusMetrics) Export(cfg config.Metrics) {
```

Comment:
Why is this deprecated?

Reply:
I thought it'd get deprecated but it didn't. Removed the `DEPRECATED` tag